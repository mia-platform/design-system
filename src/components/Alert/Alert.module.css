/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

.alert {
  border-width: 0;
  border-radius: 0;
  border-left-width: 40px;
  padding: var(--spacing-padding-lg, 16px);
  gap: var(--spacing-gap-lg, 16px);
  align-items: center;

  background-color: var(--palette-purple-100, #EDE7FF);
  border-color: var(--palette-purple-500, #986DF1);

  & > svg {
    position: absolute;
    align-self: center;
    left: calc(var(--spacing-gap-sm, 8px) - 40px);

    color: var(--palette-purple-100, #EDE7FF) !important;
  }

  &:global(.mia-platform-alert-with-description) :global(.mia-platform-alert-message) {
    margin-bottom: var(--spacing-padding-xs, 4px) !important;
  }

  & :global(.mia-platform-alert-close-icon) {
    color: var(--palette-action-secondary-contrastText, #636363);
  }

  & .titleContainer svg {
    color:  var(--palette-purple-500, #986DF1) !important;
  }

}

.compressed {
  border-left-width: 0;
  padding: 0 16px;
}

.titleContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: var(--spacing-gap-xs, 4px);
}

.compressed {
  border-left-width: 0;
  border-radius: var(--shape-border-radius-lg, 8px);
  padding: var(--spacing-padding-md, 12px);

  &:global(.mia-platform-alert-with-description) :global(.mia-platform-alert-message) {
    padding-bottom: var(--spacing-padding-sm, 8px);
  }

  & .title {
    font-size: var(--typography-bodySBold-fontSize, 14px);
    line-height: var(--typography-bodySBold-lineHeight, 16px);
    font-weight: var(--typography-bodySBold-fontWeight, 600);
    color: var(--typography-bodySBold-color, #636363);
  }

}

.title {
  font-size: var(--typography-bodyMBold-fontSize, 16px);
  line-height: var(--typography-bodyMBold-lineHeight, 24px);
  font-weight: var(--typography-bodyMBold-fontWeight, 600);
  color: var(--typography-bodyMBold-color, #636363);
}

.description {
  font-size: var(--typography-bodyS-fontSize, 14px);
  line-height: var(--typography-bodyS-lineHeight, 16px);
  font-weight: var(--typography-bodyS-fontWeight, 400);
  color: var(--typography-bodyS-color, #636363);
}

.content {

}

.info {
  background-color:  var(--palette-parimary-100, #E5F0FC);
  border-color: var(--palette-primary-500, #1890FF);
  & > svg {
    color: var(--palette-parimary-100, #E5F0FC) !important;
  }
}

.success {
  background-color:  var(--palette-success-100, #DEF8D8);
  border-color: var(--palette-success-500, #4CB61B);
  & > svg {
    color: var(--palette-success-100, #DEF8D8) !important;
  }
}

.warning {
  background-color:  var(--palette-warning-50, #FFF8D4);
  border-color: var(--palette-warning-400, #F68F1F);
  & > svg {
    color: var(--palette-warning-50, #FFF8D4) !important;
  }
}

.error {
  background-color:  var(--palette-error-100, #FFEBEB);
  border-color: var(--palette-error-500, #FF5453);
  & > svg {
    color:  var(--palette-error-100, #FFEBEB) !important;
  }
}

