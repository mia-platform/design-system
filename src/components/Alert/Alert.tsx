/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  PiCheckCircleFill,
  PiCircleHalfTiltFill,
  PiInfoFill,
  PiWarningDiamondFill,
  PiX,
  PiXSquareFill,
} from 'react-icons/pi'
import { ReactElement, useMemo } from 'react'
import { Alert as AntAlert } from 'antd'
import classnames from 'classnames'

import { Button } from '../Button/Button.tsx'
import { AlertProps } from './props.ts'
import { Icon } from '../Icon'
import { IconComponent } from '../Icon/Icon.props.ts'
import { Type } from './types.ts'
import styles from './Alert.module.css'

export const defaults = {
  isClosable: true,
}

const getIcon = (type?: Type): IconComponent => {
  switch (type) {
  case Type.Info:
    return PiInfoFill
  case Type.Warning:
    return PiWarningDiamondFill
  case Type.Error:
    return PiXSquareFill
  case Type.Success:
    return PiCheckCircleFill
  default:
    return PiCircleHalfTiltFill
  }
}

export const Alert = ({
  title: titleProp,
  description,
  isClosable = defaults.isClosable,
  type,
  children,
  isCompressed,
  icon,
  action,
  onClose,
}: AlertProps): ReactElement => {
  const className = useMemo(() => classnames([
    styles.alert,
    type && styles[type],
    isCompressed && styles.compressed,
  ]), [isCompressed, type])

  const title = useMemo(() => {
    return isCompressed
      ? (
        <div className={styles.titleContainer}>
          <Icon component={icon || getIcon(type)} size={16} />
          <span className={styles.title}>{titleProp}</span>
        </div>
      )
      : (
        <span className={styles.title}>{titleProp}</span>
      )
  }, [icon, isCompressed, titleProp, type])

  const content = useMemo(() => {
    return (description || children) && (
      <div>
        {description && (
          <span className={styles.description}>{description}</span>
        )}
        {children && (
          <div className={styles.content}>{children}</div>
        )}
      </div>
    )
  }, [children, description])

  const closable = useMemo(() => {
    return !isCompressed && isClosable && {
      closeIcon: (
        <Button
          hierarchy={Button.Hierarchy.Neutral}
          icon={<Icon component={PiX} size={16} />}
          type={Button.Type.Ghost}
        />
      ),
    }
  }, [isCompressed, isClosable])

  return (
    <AntAlert
      action={action}
      className={className}
      closable={closable}
      description={content}
      icon={<Icon component={icon || getIcon(type)} />}
      message={title}
      showIcon={!isCompressed}
      type={type}
      onClose={onClose}
    />
  )
}

Alert.Type = Type
