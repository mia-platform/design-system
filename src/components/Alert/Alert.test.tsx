/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { PiStar } from 'react-icons/pi'

import { Alert } from './Alert'
import { Button } from '../Button'

describe('Alert Snapshot Tests', () => {
  it('renders Special correctly', () => {
    const { asFragment } = render(<Alert description="Description" title="Title" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Info correctly', () => {
    const { asFragment } = render(<Alert description="Description" title="Title" type={Alert.Type.Info} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Warning correctly', () => {
    const { asFragment } = render(<Alert description="Description" title="Title" type={Alert.Type.Warning} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Error correctly', () => {
    const { asFragment } = render(<Alert description="Description" title="Title" type={Alert.Type.Error} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Success correctly', () => {
    const { asFragment } = render(<Alert description="Description" title="Title" type={Alert.Type.Success} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders InfoCompressed correctly', () => {
    const { asFragment } = render(
      <Alert
        description="Description"
        isCompressed={true}
        title="Title"
        type={Alert.Type.Info}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders WarningCompressed correctly', () => {
    const { asFragment } = render(
      <Alert
        description="Description"
        isCompressed={true}
        title="Title"
        type={Alert.Type.Warning}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders ErrorCompressed correctly', () => {
    const { asFragment } = render(
      <Alert
        description="Description"
        isCompressed={true}
        title="Title"
        type={Alert.Type.Error}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders SuccessCompressed correctly', () => {
    const { asFragment } = render(
      <Alert
        description="Description"
        isCompressed={true}
        title="Title"
        type={Alert.Type.Success}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders SpecialCompressed correctly', () => {
    const { asFragment } = render(
      <Alert description="Description" isCompressed={true} title="Title" />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders WithAction correctly', () => {
    const { asFragment } = render(
      <Alert
        action={<Button>Action</Button>}
        description="Description"
        title="Title"
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders WithCustomIcon correctly', () => {
    const { asFragment } = render(
      <Alert description="Description" icon={PiStar} title="Title" />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders WithCustomContent correctly', () => {
    const { asFragment } = render(
      <Alert isCompressed={true}>
        <ul style={{ margin: '8px 24px' }}>
          <li>Custom content</li>
          <li>Custom content</li>
        </ul>
      </Alert>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onClose if the close button is pressed', async() => {
    const onClose = jest.fn()
    render(
      <Alert
        description="Description"
        title="Title"
        onClose={onClose}
      />
    )

    screen.logTestingPlaygroundURL()
    const closeButton = screen.getByRole('img')
    fireEvent.click(closeButton)

    await waitFor(() => {
      expect(onClose).toHaveBeenCalled()
    }, { timeout: 15000 })
  })
})
