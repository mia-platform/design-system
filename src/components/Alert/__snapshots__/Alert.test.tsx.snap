// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshot Tests renders Error correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-error ant-alert-with-description alert error css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM165.66,154.34a8,8,0,0,1-11.32,11.32L128,139.31l-26.34,26.35a8,8,0,0,1-11.32-11.32L116.69,128,90.34,101.66a8,8,0,0,1,11.32-11.32L128,116.69l26.34-26.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders ErrorCompressed correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-error ant-alert-with-description ant-alert-no-icon alert error compressed css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div
          class="titleContainer"
        >
          <svg
            color="currentColor"
            fill="currentColor"
            height="16"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            style="color: currentColor;"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM165.66,154.34a8,8,0,0,1-11.32,11.32L128,139.31l-26.34,26.35a8,8,0,0,1-11.32-11.32L116.69,128,90.34,101.66a8,8,0,0,1,11.32-11.32L128,116.69l26.34-26.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
            />
          </svg>
          <span
            class="title"
          >
            Title
          </span>
        </div>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders Info correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description alert info css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders InfoCompressed correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description ant-alert-no-icon alert info compressed css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div
          class="titleContainer"
        >
          <svg
            color="currentColor"
            fill="currentColor"
            height="16"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            style="color: currentColor;"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
            />
          </svg>
          <span
            class="title"
          >
            Title
          </span>
        </div>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders Special correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description alert css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24ZM40,128A88,88,0,0,1,190.2,65.8L65.8,190.2A87.76,87.76,0,0,1,40,128Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders SpecialCompressed correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description ant-alert-no-icon alert compressed css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div
          class="titleContainer"
        >
          <svg
            color="currentColor"
            fill="currentColor"
            height="16"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            style="color: currentColor;"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24ZM40,128A88,88,0,0,1,190.2,65.8L65.8,190.2A87.76,87.76,0,0,1,40,128Z"
            />
          </svg>
          <span
            class="title"
          >
            Title
          </span>
        </div>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders Success correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-success ant-alert-with-description alert success css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm45.66,85.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders SuccessCompressed correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-success ant-alert-with-description ant-alert-no-icon alert success compressed css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div
          class="titleContainer"
        >
          <svg
            color="currentColor"
            fill="currentColor"
            height="16"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            style="color: currentColor;"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm45.66,85.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
            />
          </svg>
          <span
            class="title"
          >
            Title
          </span>
        </div>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders Warning correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-warning ant-alert-with-description alert warning css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M235.33,116.72,139.28,20.66a16,16,0,0,0-22.56,0l-96,96.06a16,16,0,0,0,0,22.56l96.05,96.06h0a16,16,0,0,0,22.56,0l96.05-96.06a16,16,0,0,0,0-22.56ZM120,80a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders WarningCompressed correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-warning ant-alert-with-description ant-alert-no-icon alert warning compressed css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div
          class="titleContainer"
        >
          <svg
            color="currentColor"
            fill="currentColor"
            height="16"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            style="color: currentColor;"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M235.33,116.72,139.28,20.66a16,16,0,0,0-22.56,0l-96,96.06a16,16,0,0,0,0,22.56l96.05,96.06h0a16,16,0,0,0,22.56,0l96.05-96.06a16,16,0,0,0,0-22.56ZM120,80a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
            />
          </svg>
          <span
            class="title"
          >
            Title
          </span>
        </div>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders WithAction correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description alert css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24ZM40,128A88,88,0,0,1,190.2,65.8L65.8,190.2A87.76,87.76,0,0,1,40,128Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-alert-action"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-5wsri9 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid button buttonMd"
        type="button"
      >
        <div
          class="buttonText"
        >
          Action
        </div>
      </button>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders WithCustomContent correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description ant-alert-no-icon alert compressed css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div
          class="titleContainer"
        >
          <svg
            color="currentColor"
            fill="currentColor"
            height="16"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            style="color: currentColor;"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24ZM40,128A88,88,0,0,1,190.2,65.8L65.8,190.2A87.76,87.76,0,0,1,40,128Z"
            />
          </svg>
          <span
            class="title"
          />
        </div>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <div
            class="content"
          >
            <ul
              style="margin: 8px 24px;"
            >
              <li>
                Custom content
              </li>
              <li>
                Custom content
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert Snapshot Tests renders WithCustomIcon correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description alert css-dev-only-do-not-override-5wsri9"
    data-show="true"
    role="alert"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M239.18,97.26A16.38,16.38,0,0,0,224.92,86l-59-4.76L143.14,26.15a16.36,16.36,0,0,0-30.27,0L90.11,81.23,31.08,86a16.46,16.46,0,0,0-9.37,28.86l45,38.83L53,211.75a16.38,16.38,0,0,0,24.5,17.82L128,198.49l50.53,31.08A16.4,16.4,0,0,0,203,211.75l-13.76-58.07,45-38.83A16.43,16.43,0,0,0,239.18,97.26Zm-15.34,5.47-48.7,42a8,8,0,0,0-2.56,7.91l14.88,62.8a.37.37,0,0,1-.17.48c-.18.14-.23.11-.38,0l-54.72-33.65a8,8,0,0,0-8.38,0L69.09,215.94c-.15.09-.19.12-.38,0a.37.37,0,0,1-.17-.48l14.88-62.8a8,8,0,0,0-2.56-7.91l-48.7-42c-.12-.1-.23-.19-.13-.5s.18-.27.33-.29l63.92-5.16A8,8,0,0,0,103,91.86l24.62-59.61c.08-.17.11-.25.35-.25s.27.08.35.25L153,91.86a8,8,0,0,0,6.75,4.92l63.92,5.16c.15,0,.24,0,.33.29S224,102.63,223.84,102.73Z"
      />
    </svg>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <span
          class="title"
        >
          Title
        </span>
      </div>
      <div
        class="ant-alert-description"
      >
        <div>
          <span
            class="description"
          >
            Description
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-alert-close-icon"
      tabindex="0"
      type="button"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
